frontend/src/utilities/js-messages.ts

```diff
@@ -0,0 +1,256 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable camelcase */
+/* eslint-disable max-classes-per-file */
+
+import { Transform, Type } from "class-transformer";
+
+export class JsMessage {
+	static responseMarker = true;
+}
+
+export class UpdateOpenDocumentsList extends JsMessage {
+	@Transform(({ value }) => value.map((tuple: [string, boolean]) => ({ name: tuple[0], isSaved: tuple[1] })))
+	open_documents!: { name: string; isSaved: boolean }[];
+}
+
+const To255Scale = Transform(({ value }) => value * 255);
+export class Color {
+	@To255Scale
+	red!: number;
+
+	@To255Scale
+	green!: number;
+
+	@To255Scale
+	blue!: number;
+
+	alpha!: number;
+
+	toRgb() {
+		return { r: this.red, g: this.green, b: this.blue, a: this.alpha };
+	}
+
+	toString() {
```

Comment:
`toRgbaCSS()` might be slightly preferable to imply what kind of string it is: a CSS RGBA string.